src/riak_kv_pb_timeseries.erl

```diff
@@ -127,6 +130,53 @@ process(#tsputreq{table = Bucket, columns = _Columns, rows = Rows}, State) ->
             {reply, missing_helper_module(Bucket, BucketProps), State}
     end;
 
+%% @ignore DELETE (as if)
+process(#tsdelreq{table = Table, key = CompoundKey_,
+                  vclock = VClock_, timeout = Timeout},
+        State) ->
+    Options =
+        lists:flatmap(
+          fun({_, undefined}) -> [];
+             ({_, default})   -> [];
+             ({K, V})         -> [{K,V}] end,
+          [{timeout, Timeout}]),
+    VClock = if VClock_ =:= undefined -> undefined;
+                %% this will trigger a get in riak_kv_delete:delete to
+                %% retrieve the actual vclock
+                el/=se -> riak_object:decode_vclock(VClock_)
+                %% else, clients may have it already (e.g., from an
+                %% earlier riak_object:get), which will short-circuit
+                %% to avoid a separate get
+             end,
+
+    CompoundKey = riak_pb_ts_codec:decode_cells(CompoundKey_),
+
+    Mod = riak_ql_ddl:make_module_name(Table),
+    try Mod:get_ddl() of
+        DDL ->
+            PKLK = make_ts_keys(CompoundKey, DDL),
+            %% = {PK, LK} = Key to pass to riak_client:delete
+            DeleteF =
```

Comment:
Instead of creating an anonymous function here, and then immediately calling it in the case statement, it would seem simpler to just say something like `Result = riak_client:delete_vclock(...` followed by `case Result of`.